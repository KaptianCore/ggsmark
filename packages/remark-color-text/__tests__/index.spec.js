import remark from 'remark'
import html from 'remark-html'
import color from '..'
import dedent from 'dedent'

describe('color text with inline style', () => {
  test('should color single line', () => {
    // Arrange
    let string = dedent`
    !# red (this is inline!)
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should not color if token never ends', () => {
    // Arrange
    let string = dedent`
    !# red
    
    this should not be red

    there's no end!!!
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color multi line', () => {
    // Arrange
    let string = dedent`
    !# red
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('multiple lines of coloring', () => {
    // Arrange
    let string = dedent`
    # Hello world before color

    before color

    !# red
    Hello I should be in red text right under the start point
    
    # I should still be red text

    !#

    text in between

    !# blue
    Hello I should be in blue text right under the start point
    
    # I should still be blue text

    more text
    I'm right above the color token and should still be blue
    !#
    I'm right under the end token
    after text

    ## Some more headings
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })
})
