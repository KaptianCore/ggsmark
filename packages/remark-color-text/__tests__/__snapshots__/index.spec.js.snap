// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color text with inline style multiple lines of coloring 1`] = `
"<h1>Hello world before color</h1>
<p>before color</p>
<div style=\\"color: red\\"><p>Hello I should be in red text right under the start point</p><h1>I should still be red text</h1></div>
<p>text in between</p>
<div style=\\"color: blue\\"><p>Hello I should be in blue text right under the start point</p><h1>I should still be blue text</h1><p>more text
I'm right above the color token and should still be blue</p></div>
<p>I'm right under the end token
after text</p>
<h2>Some more headings</h2>
"
`;

exports[`color text with inline style should color multi line 1`] = `
"<div style=\\"color: red\\">
<p>1233</p></div>
"
`;

exports[`color text with inline style should color single line 1`] = `
"<p>!# red (this is inline!)</p>
"
`;

exports[`color text with inline style should not color if token never ends 1`] = `
"<p>!# red</p>
<p>this should not be red</p>
<p>there's no end!!!</p>
"
`;
